internal/gatewayapi/extensionserverpolicy.go

```diff
@@ -201,5 +201,25 @@ func (t *Translator) translateExtServerPolicyForGateway(
 		})
 		found = true
 	}
+	for _, currListener := range gwIR.TCP {
```

Comment:
should this be flipped to ?
1. find the listeners from the `gateway` arg that are being targeted by this policy
2. find the IR for that listener
3. append ExtensionRefs to it

Reply:
WDYM?

This is already how the function works.

`gwIR` is the IR representation for the gateway, and the function goes over all of the possible listeners searching for listeners that are being targeted by this policy. Gateways have listeners in three places (`HTTP`, `TCP`, and `UDP`) so there are three identical loops here.

In each loop the policy is appended if it targets the current listener. If the policy doesn't have a `SectionName` then all the listeners of the gateway are targeted - otherwise the `SectionName` must match the listener's name.

Reply:
I was hoping we could avoid parsing o/p (IR) and instead find the IR, this would avoid steps such as `currListener.Name[strings.LastIndex(currListener.Name, "/")+1:]`, but it looks like we are already doing a similar parse in other places https://github.com/envoyproxy/gateway/blob/8a4b13dd19bd03adcf9f5c98ee249c5da7b17c17/internal/gatewayapi/backendtrafficpolicy.go#L369 but wrapping the `find` in a helper function . I'm going to resolve this comment because this code is following existing patterns